__tests__/main.test.ts

```diff
@@ -0,0 +1,235 @@
+jest.mock('fs');
+jest.mock('os');
+
+const mockedFs = require('fs');
+const mockedOs = require('os');
+const mockedToolCache = require('@actions/tool-cache');
+const mockedToolRunner = require('@actions/exec/lib/toolrunner');
+const mockedCore = require('@actions/core');
+const mockedHttpClient = require('typed-rest-client/HttpClient');
+const testUtil = require('./testUtil');
+
+const env = process.env;
+process.env = {
+    'GITHUB_WORKSPACE': 'test',
+    'GITHUB_SHA': 'sha',
+    'GITHUB_REPOSITORY': 'test_repo',
+    'GITHUB_TOKEN': 'token',
+    'GITHUB_EVENT_PATH': 'test_event.json',
+    'GITHUB_EVENT_NAME': 'pull_request'
+};
+
+Date.now = jest.fn().mockReturnValue(123);
+
+describe('Validate inputs', () => {
```

Comment:
I like the way you broke this down into multiple test suites.

Can you have a different file for each of those suites?

Reply:
I tried doing that but there seems to be a conflict in using same names for mocked modules:
`Definitions of the following identifiers conflict with those in another file: mockedFs, mockedOs, mockedToolCache, mockedToolRunner, mockedCore, mockedHttpClient, testUtil, env`.
Also, since most of these suites contain only 2-3 tests, I thought it's better to keep them all in the same file instead of different files to avoid duplicate mocks for process environment variables, certain functions and import statements for mocked modules.

Reply:
Oh! Have you tried mocking only in the `main.test.ts` file and just having the test suites in the other files?

Reply:
I'm now initializing all the mocks in describe blocks. It worked.